frontend_tests/node_tests/compose.js

```diff
@@ -89,3 +104,74 @@ people.add(bob);
     assert.equal(message.content, 'burrito');
 
 }());
+
+
+(function test_msg_server() {
+    var timeout_callback;
+
+    global.patch_builtin('setTimeout', function (f, interval) {
+        assert.equal(interval, 15000);
+        timeout_callback = f;
+    });
+
+    var request = {client: "website",
+                   type: "stream",
+                   subject: "test",
+                   stream: "all",
+                   private_message_recipient: "AARON@zulip.com",
+                   content: "Zulip is awesome",
+                   sender_id: 1,
+                   queue_id: "1485357784:0",
+                   to: "all"};
+
+    var id = 100.00;
+    var transmit_success_callback;
+    compose.transmit_message = function (request, success) {
+        transmit_success_callback = success;
+    };
+
+    compose.send_message_success = function (local_id) {
+        compose.msg_server_status[local_id] = "delivered";
+        clearTimeout(compose.delivery_timeout);
+    };
+    compose._send_message.local_id = 100;
+    var message_unsent = false;
+
+    compose.report_content_not_sent = function () {
+        message_unsent = true;
+    };
+
+    echo.message_send_error = function (local_id) {
+        clearTimeout(compose.delivery_timeout);
+        var request = {local_id: local_id};
+        compose.report_content_not_sent(request);
+    };
```

Comment:
Instead of stubbing out the entire implementation of `echo.message_send_error`, can't we stub out `clearTimeout` instead?  Then we can verify that `echo.message_send_error` calls `clearTimeout` as expected.

You may want to give these tests a quick study, as it's solving a similar problem: https://github.com/zulip/zulip/blob/master/frontend_tests/node_tests/typing_status.js